LBH.AdultSocialCare.Api/V1/Gateways/DayCarePackageGateways/DayCarePackageGateway.cs

```diff
@@ -27,6 +30,22 @@ public async Task<Guid> CreateDayCarePackage(Infrastructure.Entities.DayCarePack
             return entry.Entity.DayCarePackageId;
         }
 
+        public async Task<DayCarePackageDomain> UpdateDayCarePackage(DayCarePackageForUpdateDomain dayCarePackageForUpdate)
+        {
+            var dayCarePackageEntity = await _dbContext.DayCarePackages
+                .Where(dc => dc.DayCarePackageId.Equals(dayCarePackageForUpdate.DayCarePackageId))
+                .SingleOrDefaultAsync().ConfigureAwait(false);
+            if (dayCarePackageEntity == null)
+            {
+                throw new ResourceNotFoundException($"Unable to locate day care package {dayCarePackageForUpdate.DayCarePackageId.ToString()}");
+            }
+            // Map fields with auto mapper and save
+            _mapper.Map(dayCarePackageForUpdate, dayCarePackageEntity);
```

Comment:
I think we should avoid auto mapping properties when performing an update, if we manually assign properties then we have granular control over what gets updated